### PR TITLE
Upgraded to solidity 0.6

### PR DESCRIPTION
**Pull request by albertocuestacanada**
Upgraded to solidity 0.6 via openzeppelin-contracts-3.0.0-rc0

1. package.json & truffle.config.js changes.
2. disable eth-gas-reporter (doesn't work)
3. added `virtual` to all public and external functions
4. added `override` where needed
5. renamed `fallback` as `receive`
6. removed advanced contracts
7. one change regarding `push` not returning array length
8. corrected path to `Ownable`
9. implemented `ERC20Mintable` and `ERC721Mintable`, which have been removed

Also, ethlint doesn't recognize `receive` :(

### CODE REVIEW

File: contracts/token/ERC20Mintable.sol [hunk 1]
```diff
@@ -0,0 +1,25 @@
+pragma solidity ^0.6.0;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "../access/Community.sol";
+
+
+/**
+ * @dev Extension of {ERC20} that gives the owner permission to mint (create) new tokens as he sees fit.
+ */
+contract ERC20Mintable is ERC20, Community {
```
Comment (obernardovieira) [inline contracts/token/ERC20Mintable.sol]: What is the difference between this one and openzeppelin's ERC20Mintable?

**Comment by obernardovieira** [comment]
hmm, if ethlint isn't working and we don't have updates, then, it might become a problem. Might be time to consider some other linter :thinking:

File: contracts/token/ERC20Mintable.sol (hunk 1 above)
Comment (albertocuestacanada) [reply contracts/token/ERC20Mintable.sol]: ERC20Mintable (or ERC721Mintable) is gone from openzeppelin-contracts. They are refactoring them.

**Comment by albertocuestacanada** [comment]
Because I assume that a PR on ethlint so that it recognizes `receive ()` is out of the question...

**Comment by albertocuestacanada** [comment]
@obernardovieira, could we disable the linting check while we solve this ethlint issue? I'll still manually check for lint errors manually.

I need to merge this PR before I do anything else, as it changes all files.

**Comment by obernardovieira** [comment]
Yes, of course.

**Comment by albertocuestacanada** [comment]
Is this stuck?

**Comment by obernardovieira** [comment]
Nah, it's just stupid for some reasons https://travis-ci.org/github/HQ20/contracts/builds/666535557?utm_medium=notification&utm_source=github_status
Coveralls would not run without CI.

**Pull request merged**